metricbeat/helper/module.go

```diff
@@ -144,26 +152,18 @@ func (m *Module) Run(period time.Duration, b *beat.Beat) {
 	go m.publishing(b)
 
 	// Start fetching metrics
+	// The frequency is always based on the ticker interval. No delays are taken into account.
+	// Each Metricset must ensure to exit after timeout.
```

Comment:
Good idea. We should have this limit probably per metricset and not module so in case on metricset does not behave well, it does not affect others. Should we make this configurable and have a "reasonable" default value?

I prefer to do this in a second PR. There we should also introduce some metrics on this for httpprof.